tests/utils.py

```diff
@@ -41,7 +42,19 @@ def assert_msonable(obj, skip_keys=None, test_if_subclass=True):
         if key in skip_keys:
             continue
         assert d1[key] == d2[key]
-    _ = json.loads(obj.to_json(), cls=MontyDecoder)
+
+    try:
+        _ = json.loads(obj.to_json(), cls=MontyDecoder)
+    except Exception as e:
+        raise AssertionError(e)
+
+
+def assert_pickles(obj):
```

Comment:
Could you please test pickle load too?

Reply:
If comparison is hard, there is no need to check comparison. But it should be better if we can test whether pickled objects can be reloaded.

Reply:
I added pickle load to the tests.

testing equality is a bit hard unless we implement an __eq__ for all the classes we are testing pickling. Instead, I implemented comparing MSONable dictionaries if possible, and a weak sanity check for other objects. I think that is enough.

Let me know what you think.

Reply:
Thanks, that's great!